### DISCUSSION TITLE
release build for android

### DISCUSSION THREAD

**Discussion by tisis2** [AvaloniaUI/Avalonia#9083] [Q&A] [answered]
hi there,

currently i am evaluating avalonia for a future project and i wanted to try the performance difference in my example with the release build configuration. but i am no able to create a release build because the build keeps telling me that i would miss the workload for android. but i know its there since i am able to build in debug. i even tried with the hello world from the templates and keep getting the same issue. maybe someone here has an idea or can help me...

<img width="853" alt="image" src="https://user-images.githubusercontent.com/18416393/193596815-4252bf93-0a94-47cf-ac91-7086719feb2f.png">

**Comment by lukaszossowski** [comment]
Perhaps it is related to #8972 and #8989? It does build in VS but doesn't work anyway.

**Reply by maxkatz6** [reply]
It works, but seems to depend on SDK. And issue from @tisis2 isn't related here.
I.e. last week I built avalonia android project of latest avalonia nightly and 6.0.401 SDK on macOS, was working as expected.

**Reply by tisis2** [reply]
the issues are not related. for beeing able to deploy you first need to be able to build. and already the build has the issue here

**Comment by maxkatz6** [comment, marked as answer]
Have you tried
"dotnet workload install android"
or 
"dotnet workload restore"
?

See more information here 
https://docs.avaloniaui.net/tutorials/developing-for-mobile/android/setting-up-your-developer-environment-for-android

**Reply by tisis2** [reply]
sure i have done that. otherwise i would not have been able to build the debug build either. that is exactly what is strange. the workload is there...

**Reply by maxkatz6** [reply]
It should be possible to find a reason with binlog analysis. Using this tool https://msbuildlog.com/
Possibly comparing with debug build binlog.

**Reply by tisis2** [reply]
[buildtest.binlogs.zip](https://github.com/AvaloniaUI/Avalonia/files/9703391/buildtest.binlogs.zip)

i did a debug and release build directly after each other and this are the outputs

**Reply by tisis2** [reply]
im not used to analyze builds at that level... but as far as i can see the workload check that fails at the release build is at the restore phase and is not executed at the debug build at all due to a condition.

**Reply by tisis2** [reply]
i could figure out that it has something to do with aot. when i disable <RunAOTCompilation> and <AndroidEnableProfiledAot> in the project file the build works

**Reply by tisis2** [reply]
ok im sorry it really seems that it was an issue with workloads because dependent workloads were not downloaded for any reason and thats why the aot workloads were missing... workload repair and restore did not help at all... i did uninstalled all sdks and VS (the second time) and did reinstall the sdk and then VS with MAUI workloads and then after a workload restore it did work to do the release build
maybe it has something to do with the workload issue of dotnet https://github.com/dotnet/sdk/issues/28217
